src/integrant/core.cljc

```diff
@@ -8,10 +8,24 @@
             [weavejester.dependency :as dep]))
 
 (defprotocol RefLike
-  (ref-key [r] "Return the key of the reference."))
+  (ref-key [r] "Return the key of the reference.")
+  (ref-resolve [r config resolvef] "Return the resolved value."))
 
-(defrecord Ref    [key] RefLike (ref-key [_] key))
-(defrecord RefSet [key] RefLike (ref-key [_] key))
+(declare find-derived)
```

Comment:
Why the forward declaration?

Reply:
`find-derived` is called in each 'ref-resolve' below, but defined only later in the guts of logic.
Should I try to reorder and get rid of the forward declaration?

Reply:
Yep. Reordering is always preferable to forward declarations. Ideally reorder as few functions as possible.

Reply:
Done.